pkgs/applications/finance/irpf/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, stdenvNoCC, jdk11, makeWrapper, hicolor-icon-theme, fetchzip, fetchgit }:
+
+let
+  pkgbuild = fetchgit {
```

Comment:
What is the meaning of life of this?

Reply:
Thx for the review again Anderson, I was waiting for someone to ask as this has to do with the reproducibility of this derivation. There are three files I'm getting from the aur pkgbuild: copyright file, the desktop file and the desktop icon. I could forgo the first two since they are text files, as I could either hardcode them or use the proper tools like makeDesktopItem. However, the desktop icon should be hosted in a website whereas it can be fetched using the same url for at least 10 years. Having this in mind, fetching the icon from the government website seems prone to failure, as they can, and most likely will, change urls. 

I thought about creating a personal github only for those three files but it seems unecessary whereas there is already a git repo (although from aur git) with those. I'm looking for a better solution, but this is what I've got so far.

Reply:
I wish they just shipped the icon together with the program, but no.

Reply:
Perhaps we can extract the icon from the application itself. For example the one on the splash screen

![Screenshot from 2022-03-08 21-49-42](https://user-images.githubusercontent.com/53281855/157356523-a777dc9c-0985-465a-bb1d-ccb663ce546d.png)

I've tried looking around in the main jar file, however I can only find low quality 

![logo_srf](https://user-images.githubusercontent.com/53281855/157356908-8275ad80-d61d-4166-a28e-0f77bf0682a6.gif)

Reply:
It could be it, how did u get it?

Reply:
I found a 64x64 in `ppg-icones-4.0.jar/icones`.

My process was to all the contents of all the jars using `jar -xf` and then looking at all files with a .png extension (i used dolphin to easily scroll through the list)

Reply:
hm, interesting, I gonna look for a way to extract this file.

Reply:
Using unzip

```
unzip -j lib/ppgd-icones-4.0.jar icones/rfb64.png -d $out/share/icons/hicolor/64x64/apps
```

Also, is prefered that we store the icon in `share/icons/hicolor/64x64/apps` instead of just `share/pixmaps` ?

Reply:
I have no idea, I might test both. I actually ended up using jar xf but I think the unzip method is better. thx!

Reply:
Do you by chance know where that copyright notice was obtained from?

Reply:
I sincerely don't, but I can take a look into it.

Reply:
[Here](https://www.gov.br/receitafederal/pt-br/centrais-de-conteudo/download/pgd/dirpf). It is embedded in the website, any idea of how to get it?
![image](https://user-images.githubusercontent.com/29521461/157517358-1baff0cb-faea-460a-bc82-42bd7bb20f3c.png)

Reply:
Okay, I looked a bit more and I found a file `irpf.jar/IRPF-Licenses.txt` the header of it reads


> Termo Público de Cessão de Direito de Uso:
A Secretaria da Receita Federal do Brasil informa que o Programa 
Aplicativo para Preenchimento da Declaração de Ajuste Anual do 
Imposto de Renda de Pessoa Física do exercício de 2019, ano-calendário de 2018 (IRPF2019),
desenvolvido pelo SERPRO para a RFB, é objeto de propriedade intelectual da União 
e tem seu uso licenciado para o contribuinte, para ser utilizado exclusivamente 
para os fins a que se destina. Seu uso indiscriminado para finalidade 
diversa sujeita o responsável às penas da lei.

.... I dont speak Portuguese ... but when I ran it through google translate, it was more or less the same except some clear contextual translation errors also.... the year is different.

Reply:
If you prefer to use a hard-coded one, its cool and how you have it already works well.

Also, the idea is get everything from one source (as far as possible), so reading an excerpt from a website (which is bad enough) is a no go.

Im also questioning if the copyright file is needed as its embedded in their jar file also ... what purpose does it serve?

Reply:
U are being of such help that I cannot thank you enough. This small text is the same thing. I just thought that including thing would be necessary as far as licensing goes, but I guess that specifying that it is unfree might be enough, maybe I should just remove it.